integration/gdpr_test.go

```diff
@@ -161,7 +167,9 @@ var _ = Describe("GDPR", func() {
 
 			x.Close()
 
-			Eventually(gbytes.BufferReader(testState.gorouterSession.Out)).Should(gbytes.Say(`"RemoteAddr":"-"`))
+			Eventually(func() ([]byte, error) {
+				return os.ReadFile(testState.AccessLogFilePath())
+			}).Should(ContainSubstring(`"foo-agent" "-"`))
```

Comment:
Shouldn't this case be testing for `"RemoteAddr":"-"`?

Reply:
that's kind of confusing I know. so here is why this needs to look this it does now:
- The original test looked for the `stdout` log of RequestHandler, which is a json-structured log that prints
```
"RemoteAddr":"-"
```
as a key-value pair.
- With the removal of RequestHandler, all WebSocket requests are now handled via ProxyRoundTripper and httputils.ReverseProxy, as such their output goes to the regular access log - which is unstructured (no json)
- Now, there is no more key "RemoteAddr" but only the order in which [AccessLogRecord](https://github.com/cloudfoundry/gorouter/blob/main/accesslog/schema/access_log_record.go#L251) prints the values to the log
- This happens to be in the order of `<user-agent> <remote addr>` so, user-agent is logged first, then remote addr.
- If remote-addr shall not be logged due to GDPR, we will now have to check for an `"-"` after the user-agent string.
- I therefore had to add a user-agent string to the test so that I would have a string besides `"-"` to look for.

Reply:
👍 thanks!